src/amo/components/HomepageShelves/index.js

```diff
@@ -0,0 +1,107 @@
+/* @flow */
+import * as React from 'react';
+import config from 'config';
+
+import LandingAddonsCard from 'amo/components/LandingAddonsCard';
+import LoadingText from 'amo/components/LoadingText';
+import {
+  INSTALL_SOURCE_FEATURED,
+  INSTALL_SOURCE_FEATURED_COLLECTION,
+  LANDING_PAGE_EXTENSION_COUNT,
+  LANDING_PAGE_THEME_COUNT,
+} from 'amo/constants';
+import translate from 'amo/i18n/translate';
+import type { ResultShelfType } from 'amo/reducers/home';
+import type { I18nType } from 'amo/types/i18n';
+
+import './styles.scss';
+
+type InternalProps = {|
+  i18n: I18nType,
+|};
+
+type Props = {|
+  ...InternalProps,
+  loading: boolean,
+  shelves: Array<ResultShelfType>,
+|};
+
+class HomepageShelves extends React.Component<Props> {
+  render(): React.Node {
+    const { i18n, loading, shelves } = this.props;
+
+    let shelvesContent;
+
+    if (loading) {
+      shelvesContent = (
+        <div className="HomepageShelves-loading">
+          {[1, 2, 3].map((key) => {
+            return (
+              <LandingAddonsCard
+                className="HomepageShelves-loading-text"
+                key={`HomepageShelves-loading-text-${key}`}
+                header={<LoadingText width={100} />}
+                loading
+              />
+            );
+          })}
+        </div>
+      );
+    } else {
+      shelvesContent = shelves.map((shelf) => {
+        const { addons, criteria, endpoint, footer, title } = shelf;
+        const MOZILLA_USER_ID = config.get('mozillaUserId');
+        const titleStr = title.toString();
+        const header = titleStr.replace(/\s/g, '-');
+
+        const footerText =
+          footer && footer.text
+            ? footer.text
+            : i18n.sprintf(i18n.gettext('See more %(categoryName)s'), {
+                categoryName: titleStr.toLowerCase(),
+              });
+
+        const addonInstallSource =
+          endpoint === 'collections'
+            ? INSTALL_SOURCE_FEATURED_COLLECTION
+            : INSTALL_SOURCE_FEATURED;
+
+        const count =
+          endpoint === 'search-themes'
+            ? LANDING_PAGE_THEME_COUNT
+            : LANDING_PAGE_EXTENSION_COUNT;
+
+        let footerLinkHtml;
+        if (footer && footer.url) {
+          footerLinkHtml = <a href={footer.url}>{footerText}</a>;
+        }
+
+        const defaultFooterUrl =
+          endpoint === 'collections'
+            ? `/collections/${MOZILLA_USER_ID}/${criteria}/`
+            : `/search/${criteria}/`;
+
+        const footerLink = footerLinkHtml || defaultFooterUrl;
+
+        return (
+          <LandingAddonsCard
+            addonInstallSource={addonInstallSource}
+            addons={addons}
+            className={`Home-${header}`}
+            footerText={footerText}
+            footerLink={footerLink}
+            header={titleStr}
+            isTheme={endpoint === 'search-themes'}
+            key={header}
+            loading={loading}
```

Comment:
As above, `loading` will always be `false` here, so you can just leave this prop out.

Reply:
If I remove the line `loading={loading}`, I get a flow error saying `Cannot create LandingAddonsCard element because property 'loading' is missing in props but exists in Props.` Should it be left as is or updated to `loading={false}`?

Reply:
We should update `LandingAddonsCard` to make `loading` optional, which would address that issue. That would mean changing the type, and perhaps some code, but maybe just the type, as it's a boolean and therefore `undefined` should work just as well as `false`. Do you want to give that a try?